src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1589,5 +1596,20 @@ internal void ReleaseAcceptingNewBlocks()
                 }
             }
         }
+        
+        public Block[] GetInvalidBlocks()
+        {
+            IList<Block> blockList = new List<Block>(); 
+            IDictionary<long, HashSet<Keccak>> tempDict = _invalidBlocks.Clone();
+            foreach (HashSet<Keccak> invalidBlockHashes in tempDict.Values)
+            {
+                foreach (var hash in invalidBlockHashes)
+                {
+                    blockList.Add(FindBlock(hash, BlockTreeLookupOptions.None));
```

Comment:
I don't think it will work. Invalid blocks are deleted from the block tree in DeleteInvalidBlock, thus FindBlock won't find them.